pkg/apis/extensions/v1alpha1/types_containerruntime.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ Object = (*ContainerRuntime)(nil)
+
+// ExtensionResource is a constant for the name of the Extension resource.
+const ContainerRuntimeResource = "ContainerRuntime"
+const CRINameWorkerLabel = "worker.gardener.cloud/cri-name"
+const ContainerRuntimeWorkerLabel = "worker.gardener.cloud/container-runtime=%s"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ContainerRuntime is a specification for a container runtime resource.
+type ContainerRuntime struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ContainerRuntimeSpec   `json:"spec"`
+	Status            ContainerRuntimeStatus `json:"status"`
+}
+
+// GetExtensionSpec implements Object.
+func (i *ContainerRuntime) GetExtensionSpec() Spec {
+	return &i.Spec
+}
+
+// GetExtensionStatus implements Object.
+func (i *ContainerRuntime) GetExtensionStatus() Status {
+	return &i.Status
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ExtensionList is a list of Extension resources.
```

Comment:
```suggestion
// ContainerRuntimeList is a list of ContainerRuntime resources.
```

replace `Extension` with `ContainerRuntime` everywhere